pyo3-macros-backend/src/params.rs

```diff
@@ -310,12 +310,18 @@ fn impl_arg_param(
 
         Ok(quote_arg_span! {
             let #mut_ _tmp: #target_ty = #arg_value_or_default?;
-            #[allow(clippy::needless_option_as_deref)]
+            #[allow(clippy::needless_option_as_deref, clippy::borrow_deref_ref)]
             let #arg_name = #borrow_tmp;
         })
-    } else {
+    } else if cfg!(feature = "pyproto") {
+        // arg.ty is kinda weird
         Ok(quote_arg_span! {
             let #arg_name = #arg_value_or_default?;
         })
+    } else {
+        let ty = arg.ty;
```

Comment:
This might need a call to `remove_lifetime`. Can you add a test for a pyfunction with something like `Vec<&'py PyAny>` as an argument? I've also been wondering if it's possible for us to copy lifetimes from the function definition onto the generated `pyfunction` - I think that might also solve the problem but not sure.